### PR TITLE
Support `address.code` in sema

### PR DESCRIPTION
**Pull request by xermicus**

### CODE REVIEW

File: src/sema/expression/member_access.rs [hunk 1]
```diff
@@ -347,37 +347,52 @@ pub(super) fn member_access(
...
-                        diagnostics.push(Diagnostic::error(
-                            expr.loc(),
-                            "polkadot can only retrieve balance of this, like 'address(this).balance'".to_string(),
-                        ));
-                        return Err(());
-                    }
+                if !is_this {
+                    diagnostics.push(Diagnostic::error(
+                        expr.loc(),
+                        "polkadot can only retrieve balance of this, like 'address(this).balance'"
```
Comment (LucasSte) [inline src/sema/expression/member_access.rs:367]:
```suggestion
                        "polkadot can only retrieve balance of 'this', like 'address(this).balance'"
```

**Review by LucasSte — APPROVED** [review]

File: src/sema/expression/member_access.rs [hunk 2]
```diff
@@ -347,37 +347,52 @@ pub(super) fn member_access(
...
+                diagnostics.push(Diagnostic::error(
+                    expr.loc(),
+                    format!("'address.code' is not supported on {}", ns.target),
+                ));
+                return Err(());
+            }
+            used_variable(ns, &expr, symtable);
+            return Ok(Expression::Builtin {
+                loc: *loc,
+                tys: vec![Type::Slice(Type::Uint(8).into())],
```
Comment (seanyoung) [inline src/sema/expression/member_access.rs:392]: Type::DynamicBytes

**Comment by seanyoung** [comment]
You also need an evm test cases without errors

**Comment by xermicus** [comment]
@seanyoung it's already covered because it decreases the failing EVM tests. But I added a bunch more, can't harm to have our own.

**Review by seanyoung — APPROVED** [review]

**Pull request merged**